R/reformat.R

```diff
@@ -83,10 +88,14 @@ reformat.character <- function(obj, format, ...) {
 #' reformat(obj, format)
 #' reformat(obj, format, .na_last = FALSE, .to_NA = "b", .drop = FALSE)
 #'
-reformat.factor <- function(obj, format, ...) {
+reformat.factor <- function(obj, format, ..., verbose = FALSE) {
   checkmate::assert_class(format, "rule")
+  checkmate::assert_flag(verbose)
 
   format <- do.call(rule, modifyList(as.list(format), list(...), keep.null = TRUE))
+  if (verbose) {
+    print(format)
```

Comment:
since format is printed, we may want to improve how formats are printed, currently it is 

```
Mapping of:
A  <-  a 
NN  <-  NA 
NA <-   
Convert to factor: TRUE 
Drop unused level: FALSE 
NA-replacing level in last position: TRUE 
```

right hand side of NA is empty, is it empty string?
and we can not distinguish between "NA" string and `NA_character`. maybe just follow the `forcats` to use `<NA>` to represent the `NA_character`?

Reply:
Good point, since it refers to another method, I suggest to treat it in another issue (https://github.com/insightsengineering/dunlin/issues/163). I have to think about how we want to go about it to avoid confusion.